alfresco-health-processor-platform/src/main/java/eu/xenit/alfresco/healthprocessor/indexing/IndexingStrategy.java

```diff
@@ -3,11 +3,24 @@
 import java.util.Set;
 import org.alfresco.service.cmr.repository.NodeRef;
 
+/**
+ * Strategy used by the Health-Processor to iterate over (a subset of) Alfresco nodes. Once {@link #getNextNodeIds(int)}
+ * returns an empty set, the Health-Processor considers the iteration done and the current cycle will be terminated.
+ * Implementations can use the {@link #onStart()} and {@link #onStart()} to (re-)initialize state or open and close
+ * additional resources.
+ *
+ * @see eu.xenit.alfresco.healthprocessor.indexing.IndexingStrategyFactoryBean
+ */
 public interface IndexingStrategy {
 
-    void reset();
+    default void onStart() {
+
```

Comment:
Just a thought:
Adding tracing/debug logging might be useful, but then needs to be called in a weird way ... 
`IndexingStrategy.super.onStart()` :roll_eyes:

Reply:
:thinking: I don't fully understand this comment.